storage/migrations/mysql/1_init.sql

```diff
@@ -87,158 +86,164 @@ CREATE TABLE IF NOT EXISTS `templates`
+    FOREIGN KEY (`template_id`) REFERENCES templates(`id`) ON DELETE SET NULL ON UPDATE CASCADE,
+    INDEX idx_id_created_at (`id`, `created_at`),
+    INDEX idx_user_id (`user_id`)
 ) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci;
 
 CREATE TABLE IF NOT EXISTS `subscribers` (
-  `id`          BIGINT UNSIGNED PRIMARY KEY AUTO_INCREMENT NOT NULL,
-  `user_id`     INTEGER UNSIGNED NOT NULL,
-  `name`        VARCHAR(191) DEFAULT NULL,
-  `email`       VARCHAR(191) NOT NULL,
-  `metadata`    JSON,
-  `blacklisted` TINYINT(1) NOT NULL,
-  `active`      TINYINT(1) NOT NULL,
-  `created_at`  DATETIME(6) NOT NULL,
-  `updated_at`  DATETIME(6) NOT NULL,
-  FOREIGN KEY (`user_id`) REFERENCES users(`id`),
-  INDEX id_created_at (`id`, `created_at`),
-  UNIQUE(`user_id`, `email`)
+    `id`          BIGINT UNSIGNED PRIMARY KEY AUTO_INCREMENT NOT NULL,
+    `user_id`     INTEGER UNSIGNED NOT NULL,
+    `name`        VARCHAR(191) DEFAULT NULL,
+    `email`       VARCHAR(191) NOT NULL,
+    `metadata`    JSON,
+    `blacklisted` TINYINT(1) NOT NULL,
+    `active`      TINYINT(1) NOT NULL,
+    `created_at`  DATETIME(6) NOT NULL,
+    `updated_at`  DATETIME(6) NOT NULL,
+    FOREIGN KEY (`user_id`) REFERENCES users(`id`),
+    INDEX idx_id_created_at (`id`, `created_at`),
+    INDEX idx_user_id (`user_id`),
+    INDEX idx_user_blacklist_active (`user_id`, `blacklisted`, `active`),
+    UNIQUE(`user_id`, `email`)
 ) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci;
 
 CREATE TABLE IF NOT EXISTS `segments` (
-  `id`          INTEGER UNSIGNED PRIMARY KEY AUTO_INCREMENT NOT NULL,
-  `user_id`     INTEGER UNSIGNED NOT NULL,
-  `name`        VARCHAR(191) NOT NULL,
-  `created_at`  DATETIME(6) NOT NULL,
-  `updated_at`  DATETIME(6) NOT NULL,
-  FOREIGN KEY (`user_id`) REFERENCES users(`id`),
-  INDEX id_created_at (`id`, `created_at`)
+    `id`          INTEGER UNSIGNED PRIMARY KEY AUTO_INCREMENT NOT NULL,
+    `user_id`     INTEGER UNSIGNED NOT NULL,
+    `name`        VARCHAR(191) NOT NULL,
+    `created_at`  DATETIME(6) NOT NULL,
+    `updated_at`  DATETIME(6) NOT NULL,
+    FOREIGN KEY (`user_id`) REFERENCES users(`id`),
+    INDEX idx_id_created_at (`id`, `created_at`),
+    INDEX idx_user_id (`user_id`)
 ) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci;
 
 CREATE TABLE IF NOT EXISTS `subscribers_segments` (
-  `segment_id`    INTEGER UNSIGNED NOT NULL,
-  `subscriber_id` BIGINT UNSIGNED NOT NULL,
-  PRIMARY KEY (`segment_id`, `subscriber_id`),
-  FOREIGN KEY (`segment_id`) REFERENCES segments(`id`),
-  FOREIGN KEY (`subscriber_id`) REFERENCES subscribers(`id`)
+    `segment_id`    INTEGER UNSIGNED NOT NULL,
+    `subscriber_id` BIGINT UNSIGNED NOT NULL,
+    PRIMARY KEY (`segment_id`, `subscriber_id`),
+    FOREIGN KEY (`segment_id`) REFERENCES segments(`id`),
+    FOREIGN KEY (`subscriber_id`) REFERENCES subscribers(`id`),
+    INDEX idx_segment_id (`segment_id`),
+    INDEX idx_subscriber_id (`subscriber_id`)
```

Comment:
Same here, no need for indexes. I'll skip the rest of the comments below make, just make sure to remove them @gudgl.

Reply:
done